unit-test/background/reference-tests/broken-site-reporting-tests.js

```diff
@@ -64,7 +64,7 @@ for (const setName of Object.keys(testSets)) {
 
                 if (test.expectReportURLParams) {
                     test.expectReportURLParams.forEach(param => {
-                        expect(requestURLString).toContain(`${param.name}=${param.value}`)
+                        expect(requestURLString).toMatch(new RegExp(`[?&]${param.name}=${param.value}[&$]`))
```

Comment:
Can we use the URL constructor here?

Untested but something like this:
```suggestion
                        let url = new URL(requestURLString)
                        expect(url.searchParams.get(param.name)).equals(param.value)
```

Reply:
I used that initially but URL object automatically decodes URL params and I get test failures:

> Expected 'https://example.test/' to be 'https%3A%2F%2Fexample.test%2F'

I wanted tests to check if parameters are properly url encoded, so I don't want to adjust tests. I guess I could encode those parameters that I will get back again, but this is not great either.

Reply:
Ah makes sense, I still think I'd prefer using encodeURIComponent if that works than RegExp.

Reply:
Bleh, encodeURIComponent behaves differently than encodeURI and, unexpectedly, encodes `,`. I'll have to find a different way.